packages/core/src/lib/toast/toast-banner.svelte

```diff
@@ -0,0 +1,119 @@
+<svelte:options immutable />
+
+<script lang="ts">
+import cx from 'classnames';
+import { createEventDispatcher } from 'svelte';
+import { Button, Icon, type IconName } from '$lib';
+import { ToastVariant, type ToastVariantType } from './variants';
+
+/** The type of toast banner to be displayed to the user */
+export let variant: ToastVariantType;
+
+/** Whether there is a close icon displayed */
+export let closeable: boolean | undefined = false;
+
+/** Whether there is an action button displayed */
+export let action = '';
+
+/** The scaling applied on the x axis of the page*/
+export let progress = 1;
+
+/** Additional CSS classes to pass to the banner. */
+let extraClasses: cx.Argument = '';
+export { extraClasses as cx };
+
+const dispatch = createEventDispatcher<{
+  /** when the toast close button is pressed*/
+  close: null;
+  /**  when the action button is pressed*/
+  action: null;
+}>();
+
+const handleClose = () => dispatch('close');
+
+const handleAction = () => dispatch('action');
```

Comment:
I know this was copied over from the the notification code, but can we use function props rather than events? Svelte itself is moving away from events, and in my experience, events are annoying to test for and statically analyze